be/src/runtime/exec_env_init.cpp

```diff
@@ -170,6 +171,7 @@ Status ExecEnv::_init(const std::vector<StorePath>& store_paths) {
     _block_spill_mgr = new BlockSpillManager(_store_paths);
     _file_meta_cache = new FileMetaCache(config::max_external_file_meta_cache_num);
     _memtable_memory_limiter = std::make_unique<MemTableMemoryLimiter>();
+    _wal_manager = new WalManager(this, config::group_commit_replay_wal_dir);
```

Comment:
use smart pointer, not use raw pointer.
Add ENABLE_FACTORY_CREATOR to all classes.